perfscale_regression_ci/scripts/encryption/toggle_encryption.sh

```diff
@@ -0,0 +1,151 @@
+## To enable: ./toggle_encryption.sh enable 50
+## To disable: ./toggle_encryption.sh disable 50
+###########################################################################################################
+
+function log {
+    echo -e "$*"
+}
+
+# Setting variables and names of new created secrets.
+mode=""
+timeout=$(($2 * 60))
+## Temporary files to store secrets
+OC_SECRET="/tmp/openshift-apiserver-secret.yaml"
+KB_SECRET="/tmp/openshift-kube-apiserver-secret.yaml"
+OAUTH_SECRET="/tmp/openshift-oauth-apiserver-secret.yaml"
+OC_MIGRATED_RESOURCES="'{\"resources\":[{\"Group\":\"route.openshift.io\",\"Resource\":\"routes\"}]}'"
+## Resources A and B are the the same json objects, but look different as a strings (in check_secret I'm comparing string from yaml)
+KB_MIGRATED_RESOURCES_A="'{\"resources\":[{\"Group\":\"\",\"Resource\":\"configmaps\"},{\"Group\":\"\",\"Resource\":\"secrets\"}]}'"
+KB_MIGRATED_RESOURCES_B="'{\"resources\":[{\"Group\":\"\",\"Resource\":\"secrets\"},{\"Group\":\"\",\"Resource\":\"configmaps\"}]}'"
+OAUTH_MIGRATED_RESOURCES_A="'{\"resources\":[{\"Group\":\"oauth.openshift.io\",\"Resource\":\"oauthaccesstokens\"},{\"Group\":\"oauth.openshift.io\",\"Resource\":\"oauthauthorizetokens\"}]}'"
+OAUTH_MIGRATED_RESOURCES_B="'{\"resources\":[{\"Group\":\"oauth.openshift.io\",\"Resource\":\"oauthauthorizetokens\"},{\"Group\":\"oauth.openshift.io\",\"Resource\":\"oauthaccesstokens\"}]}'"
+## Getting number of existing secrets.
+oc_secrets_num=$(oc get secrets -n openshift-config-managed | grep -c encryption-key-openshift-apiserver)
+kb_secrets_num=$(oc get secrets -n openshift-config-managed | grep -c encryption-key-openshift-kube-apiserver)
+oauth_secrets_num=$(oc get secrets -n openshift-config-managed | grep -c encryption-key-openshift-oauth-apiserver)
+## Setting names for secrets created during encryption/decryption.
+((oc_secrets_num = oc_secrets_num + 1))
+((kb_secrets_num = kb_secrets_num + 1))
+((oauth_secrets_num = oauth_secrets_num + 1))
+oc_secrets_name=$(echo -e "encryption-key-openshift-apiserver-${oc_secrets_num}")
+kb_secrets_name=$(echo -e "encryption-key-openshift-kube-apiserver-${kb_secrets_num}")
+oauth_secrets_name=$(echo -e "encryption-key-openshift-oauth-apiserver-${oauth_secrets_num}")
+log "Secrets to check:"
+log "${oc_secrets_name}"
+log "${kb_secrets_name}"
+log "${oauth_secrets_name}"
+
+function enable_encryption {
+    mode="aescbc"
+    oc patch apiserver/cluster -p '{"spec":{"encryption": {"type":"aescbc"}}}' --type merge
+    log "Encryption enabled"
+}
+
+function disable_encryption {
+    mode="identity"
+    oc patch apiserver/cluster -p '{"spec":{"encryption": {"type":"identity"}}}' --type merge
+    log "Encryption disabled"
+}
+
+function check_secret {
+    secret_file="$1"
+    migrated_resources_value_a="$2"
+    migrated_resources_value_b="$3"
+    secret_migrated_resources=$(grep "encryption.apiserver.operator.openshift.io/migrated-resources" "${secret_file}" | awk '{print $2}')
```

Comment:
Are these messages in your output expected for this command? 

```
06-06 14:09:41.236  Missing: encryption.apiserver.operator.openshift.io/migrated-resources in /tmp/openshift-apiserver-secret.yaml secret.
06-06 14:09:41.236  Missing: encryption.apiserver.operator.openshift.io/migrated-timestamp in /tmp/openshift-apiserver-secret.yaml secret.
```

Reply:
This lines are from comparison L:69, L:76 L:83
Should be log message simpler or show nothing here?

Reply:
I think if it prints missing it is misleading

Reply:
Changed to `Still waiting for`